_posts/2019-03-28-Add-Files-From-Git-Repos.md

```diff
@@ -0,0 +1,150 @@
+repo and composer will make an rpm of it and install the files under the
+destination path that you specify.  The rpm that is installed in the image
+includes the exact details about where the files came from so there is no
+question about their source.
+
+Let's walk through an example scenario with nginx and some virtual hosts.
+
+Nginx virtual hosts
+-------------------
+
+You can easily setup virtual hosts in nginx by adding configuration files to the
+`/etc/nginx/conf.d/` directory.  In their simplest form they look like this:
+
+    server {
+            listen 80;
+            server_name one-site.myhost.com;
+            root /var/www/one-site.myhost.com/public_html;
+            index index.php index.html;
+    }
+
+I have setup an [example git repository here](https://github.com/bcl/extra-files), it includes 3 vhosts
+along with their html directories under `/var/www/`.
+
+Starting nginx at boot time
+---------------------------
+
+One of the limitations of lorax-composer is that there is currently no way to
+specify which services need to be started at boot time -- we're working on
+this, take a look at [my customizations proposal
+PR](https://github.com/weldr/lorax/pull/634) for an idea of where we're headed.
+In order to start a service it either needs to be setup to start by the package
+that installs it, or enabled by using something like Ansible after the system
+boots.  The other alternative is to drop in a symlink to its systemd service
+file as part of our extra files repository.
+
+To do this you create a symlink under `/etc/systemd/system/multi-user.target.wants/`
+that points to the service file.  This is relative to the top directory of the
+git repository, NOT the / of your development system::
+
+    mkdir -p etc/systemd/system/multi-user.target.wants
+    cd etc/systemd/system/multi-user.target.wants
+    ln -s /usr/lib/systemd/system/nginx.service .
+
+Building an nginx qcow2 image with the extra files
+--------------------------------------------------
+
+We now have a git repository with all the files we want to add, as well as a symlink that
+will cause nginx to start at boot time. We need to create a blueprint, from the command
+line, that includes nginx and the git repository information.
+
+Open up a text editor and create `nginx-extra-vhosts.toml` file and add this to it:
+
+    name = "nginx-extra-vhosts"
+    description = "An nginx server including some custom vhosts"
+    version = "0.0.1"
+    groups = []
+
+    [[packages]]
+    name = "nginx"
+    version = "1.*"
+
+    [[packages]]
+    name = "nginx-all-modules"
+    version = "1.*"
+
+    [[customizations.user]]
+    name = "root"
+    key = "ssh-rsa ..... user@localhost.localdomain"
+
+    [[repos.git]]
+    rpmname = "vhosts"
+    rpmversion = "1.0"
+    rpmrelease = "1"
+    summary = "vhosts for nginx server"
+    repo = "https://github.com/bcl/extra-files"
+    ref = "v1.0"
```

Comment:
So `ref` specifies which commit in the repo we are pulling the files from?
What's the purpose of `rpmversion` and `rpmrelease`?

Reply:
Correct. rpmversion and release are used to create the rpm file which is installed on the image. I'll add a better description of that.